### PR TITLE
chore(service-disco): add component tests

### PR DESCRIPTION
**Pull request by gmelodie**
Summary

  Adds an integration test suite that exercises the full service-discovery protocol across real networked
  nodes — register, get-ads, and end-to-end advertise→discover — using the advertiser, registrar, and
  discoverer components built in prior branches.

  - tests/libp2p/service_discovery/test_integration.nim (new):
    - processRetryTicket — unit tests (no I/O) covering all ticket-validation edge cases: no ticket,
  mismatched bytes, bad signature, expired/before/after-window, within-window, and sufficient-elapsed-time
  scenarios.
    - Integration - handleRegister — networked tests: first register → Wait, wrong-key ticket → Rejected,
  out-of-window ticket → Rejected, safetyParam=0 → Confirmed on first attempt.
    - Integration - handleGetAds — networked tests: empty cache returns no ads, stored ads are returned
  correctly, fReturn cap is respected.
    - Integration - end-to-end — addProvidedService triggers background registration; lookup on a separate
  node finds the advertisement.

  ---
  Affected Areas

  - Gossipsub
  - Transports
  - Peer Management / Discovery
  - Protocol Logic
  - Build / Tooling
  - Other
  Test-only change: new integration suite, no library code modified.

  ---
  Compatibility & Downstream Validation

  No library code changed. No API, ABI, or wire-format impact. Downstream consumers are unaffected.

  ---
  Impact on Library Users

  None — this PR adds tests only.

  ---
  Risk Assessment

  - Test flakiness — low risk. checkUntilTimeout guards the async synchronisation point in the end-to-end
  test; no fixed sleeps.

  ---
  References

  - #2055

### CODE REVIEW

File: tests/libp2p/service_discovery/test_integration.nim [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+      providerExpirationInterval = chronos.seconds(1),
+      republishProvidedKeysInterval = chronos.milliseconds(50),
+    ),
+    discoConfig = discoConf,
+    xprPublishing = false,
+  )
+  switch.mount(kad)
+  kad
+
+# ===========================================================================
```
Comment (richard-ramos) [inline tests/libp2p/service_discovery/test_integration.nim]: Can we remove these separators? not really needed nor help the code breathe :)

**Review by richard-ramos — APPROVED** [review]

File: tests/libp2p/service_discovery/test_integration.nim [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    # Directly populate the registrar cache with 4 distinct ads
+    for _ in 0 ..< 4:
+      let ad = makeAdvertisement(serviceName)
+      registrarNode.acceptAdvertisement(serviceId, ad)
+
+    let found = await discovererNode.lookup(serviceId)
+    check found.isOk()
+    check found.get().len <= 2
+
+suite "Integration - end-to-end":
```
Comment (vladopajic) [inline tests/libp2p/service_discovery/test_integration.nim]: `integration tests`  term is reserved for tests in `/tests/integration`.
`component tests` is when we create unit like test using many nodes.

this terminology is already established and should be followed through to avoid further confusion.

**Review by vladopajic — APPROVED** [review]
